<h2>Why is this an issue?</h2>
<p>Redundant Boolean literals should be removed from expressions to improve readability.</p>
<h3>Noncompliant code example</h3>
<pre>
if boolFunc() || false {
    // ...
}

flag := x &amp;&amp; true
</pre>
<h3>Compliant solution</h3>
<pre>
if boolFunc() {
    // ...
}

flag := x
</pre>

