<h2>Why is this an issue?</h2>
<p>For better readability, do not put more than one statement on a single line.</p>
<h3>Noncompliant code example</h3>
<pre>
foo(); bar() // Noncompliant
</pre>
<h3>Compliant solution</h3>
<pre>
foo()
bar()
</pre>
<h3>Exceptions</h3>
<p>Control flow statements with a single nested statement are ignored.</p>
<pre>
if condition { doSomething() } // Compliant
</pre>

