<p>For better readability, do not put more than one statement on a single line.</p>
<h2>Noncompliant Code Example</h2>
<pre>
foo(); bar() // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
foo()
bar()
</pre>
<h2>Exceptions</h2>
<p>Control flow statements with a single nested statement are ignored.</p>
<pre>
if condition { doSomething() } // Compliant
</pre>

