<h2>Why is this an issue?</h2>
<p>Using the same value on both sides of a binary operator is a code defect. In the case of logical operators, it is either a copy/paste error and,
therefore, a bug, or it is simply duplicated code and should be simplified. In the case of bitwise operators and most binary mathematical operators,
having the same value on both sides of an operator yields predictable results and should be simplified as well.</p>
<h3>Noncompliant code example</h3>
<pre>
func main() {
  v1 := (true &amp;&amp; false) &amp;&amp; (true &amp;&amp; false) // Noncompliant
}
</pre>
<h3>Compliant solution</h3>
<pre>
func main() {
  v1 := (true &amp;&amp; false) // Compliant
}
</pre>
<h3>Exceptions</h3>
<p>This rule ignores <code>*</code>, <code>+</code>, <code>&lt;&lt;</code> and <code>=</code>.</p>
<h2>Resources</h2>
<ul>
  <li> {rule:go:S1656} - Implements a check on <code>=</code>. </li>
</ul>

