<h2>Why is this an issue?</h2>
<p>Unused parameters are misleading. Whatever the values passed to such parameters, the behavior will be the same.</p>
<h3>Noncompliant code example</h3>
<pre>
void doSomething(Integer a, Integer b) {     // "b" is unused
  compute(a);
}
</pre>
<h3>Compliant solution</h3>
<pre>
void doSomething(Integer a) {
  compute(a);
}
</pre>

