<h2>Why is this an issue?</h2>
<p>Nested <code>if</code>, <code>for</code>, <code>while</code>, <code>switch</code>, and <code>try</code> statements are key ingredients for making
whatâ€™s known as "Spaghetti code".</p>
<p>Such code is hard to read, refactor and therefore maintain.</p>
<h3>Noncompliant code example</h3>
<p>With the default threshold of 3:</p>
<pre>
switch on i { // Compliant - depth = 1
  when 1 {
    /* ... */
    for (Integer i = 0, j = 0; i &lt; 10; i++) { // Compliant - depth = 2
      /* ... */
      Integer j = i + 1;
      if(j == 0){// Compliant - depth = 3, not exceeding the limit
        /* ... */
        while(j &lt; 10){ // Noncompliant - depth = 4
          /* ... */
          if(j == 1){ // Depth = 5, exceeding the limit, but issues are only reported on depth = 4
            /* ... */
          }
        }
      }
    }
  }
}
</pre>

