<h2>Why is this an issue?</h2>
<p>Using the same value on both sides of a binary operator is a code defect. In the case of logical operators, it is either a copy/paste error and,
therefore, a bug, or it is simply duplicated code and should be simplified. In the case of bitwise operators and most binary mathematical operators,
having the same value on both sides of an operator yields predictable results and should be simplified as well.</p>
<h3>Noncompliant code example</h3>
<pre>
if ( a == a ) { // always true
  foo();
}
if ( a != a ) { // always false
  foo();
}
if ( a == b &amp;&amp; a == b ) { // if the first one is true, the second one is too
  foo();
}
if ( a != b || a &lt;&gt; b ) { // if the first one is true, the second one is too
  foo();
}

Integer i = 5 / 5; // always 1
Integer j = 5 - 5; // always 0
</pre>
<h3>Exceptions</h3>
<ul>
  <li> This rule ignores <code>*</code>, <code>+</code>, <code>^</code>, <code>&amp;</code>, <code>|</code>, <code>=</code>, <code>&lt;&lt;</code>,
  <code>&gt;&gt;</code> and <code>&gt;&gt;&gt;</code>. </li>
</ul>
<h2>Resources</h2>
<ul>
  <li> {rule:apex:S1656} - Implements a check on <code>=</code>. </li>
</ul>

