<h2>Why is this an issue?</h2>
<p>By default tests will run in system mode, i.e. without taking into account users permissions. In order to be realistic, a test needs to run
Business logic code in User context. This is done by encapsulating the tested code in <a
href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_testing_tools_runas.htm"><code>System.runAs()</code></a>.</p>
<p>This rule raises an issue when a test function, i.e. a function annotated with <code>@isTest</code> or <code>testMethod</code>, does not contain a
<code>System.runAs()</code> call.</p>
<h3>Noncompliant code example</h3>
<pre>
@isTest
private class TestClass {
    @isTest
    public static void testMethod() { // NonCompliant
        // Setup test data
        User u = new User(...);
        Case c = new Case (Name = 'Test');
        Test.startTest();
            Insert c;
        Test.stopTest();
    }
}
</pre>
<h3>Compliant solution</h3>
<pre>
@isTest
private class TestClass {
    @isTest
    public static void testMethod() {
        // Setup test data
        User u = new User(...);
        Case c = new Case (Name = 'Test');
        System.runAs(u) {
            Test.startTest();
            Insert c;
            Test.stopTest();
        }
    }
}
</pre>
<h3>Exceptions</h3>
<p>No issue will be raised if the test class, i.e. the class annotated with <code>@isTest</code>, contains helper methods, i.e. methods
<strong>not</strong> annotated with <code>@isTest</code> or <code>testmethod</code>, which contain calls to <code>System.runAs()</code>. This
indicates that the test code has been factorized and the rule would raise false positives.</p>
<h2>Resources</h2>
<ul>
  <li> <a href="https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_testing_tools_runas.htm">Using the runAs Method</a>
  </li>
</ul>

