<h2>Why is this an issue?</h2>
<p>Using <code>getRecordTypeInfosByName()</code> requires you to pass the Record Type Name. A Record Type Name is a label, which means that it can be
easily changed by an administrator. When this happens, the code will simply stop working until a developer updates the Record Type Name. Even worse,
Labels are usually translated in multi-language environments, which will again make the code fail.</p>
<p>The method getRecordTypeInfosByDeveloperName() also enables you to get the Record Type Id, but it requires instead a Record Type API Name, which is
rarely changed.</p>
<p>Thus it is safer to use <code>getRecordTypeInfosByDeveloperName()</code> than <code>getRecordTypeInfosByName()</code>.</p>
<h3>Noncompliant code example</h3>
<pre>
Id RTId = Schema.SObjectType.Case.getRecordTypeInfosByName().get('Asia Service').getRecordTypeId();
</pre>
<h3>Compliant solution</h3>
<pre>
Id RTId = Schema.SObjectType.Case.getRecordTypeInfosByDeveloperName().get('Asia_Service').getRecordTypeId();
</pre>
<h2>Resources</h2>
<ul>
  <li> <a href="https://smukov.github.io/blog/2018/06/09/Record-Type-Id-By-Developer-Name/">Get Record Type Id by Developer Name</a> </li>
</ul>

