<h2>Why is this an issue?</h2>
<p>Moving the business logic out of Trigger functions and into dedicated "Trigger Handler" classes improves the applicationâ€™s design:</p>
<ul>
  <li> the code is easier to test and maintain. </li>
  <li> it helps avoiding some bugs such as trigger recursion. </li>
</ul>
<p>The Trigger functions should only dispatch calls to the corresponding Trigger Handler classes. See the links below for examples of Trigger Handler
designs.</p>
<p>This rule raises an issue when a Trigger function contains one of the following syntax elements: loops, switch-case, try blocks, SOQL queries, DML
queries, SOSL queries. The goal is to detect Trigger functions which have a complex logic. In practice method calls and if statements are enough to
dispatch records for processing.</p>
<h3>Noncompliant code example</h3>
<pre>
trigger MyTrigger on Account(after insert, after update) { // Noncompliant. The trigger is processing records itself instead of using a Trigger Handler.
    for(Account a : Trigger.New) {
        // ...
    }
}
</pre>
<h2>Resources</h2>
<ul>
  <li> <a href="https://web.archive.org/web/20210509202306/https://github.com/ChrisAldridge/Lightweight-Trigger-Framework">Lightweight Apex Trigger
  Framework</a> </li>
  <li> <a
  href="https://web.archive.org/web/20230130071426/https://meltedwires.com/2013/06/05/trigger-pattern-for-tidy-streamlined-bulkified-triggers-revisited/">Trigger Pattern for Tidy, Streamlined, Bulkified Triggers Revisited</a> </li>
</ul>

