<h2>Why is this an issue?</h2>
<p>Using absolute URLs to Salesforce Pages is bug prone. Different sandboxes and production environments will have different instance names (like
"na10", "na15" etc.). Code using absolute URLs will only work when it runs in the corresponding salesforce instances. It will break as soon as it is
deployed in another one. Thus only relative URLs, i.e. without the domain and subdomain names, should be used when pointing to a salesforce page.</p>
<p>This rule raises an issue when a string matches the regular expression:</p>
<p>{noformat}</p>
<p>(?&lt;!\w)(login|test|(dns|test|[a-z]{1,2})\d++)\.(salesforce|force|visual\.force|content\.force)\.com(?!\w)</p>
<p>{noformat}</p>
<h3>Noncompliant code example</h3>
<pre>
public PageReference testRedirect() {
    String strUrl ='https://na8.salesforce.com/TestVFPage?AcoountId=999'; // Noncompliant
    PageReference newUrl = new PageReference(strUrl);
    newURL.setRedirect(true);
    return newURL;
}
</pre>
<h3>Compliant solution</h3>
<pre>
public PageReference testRedirect() {
    String strUrl = URL.getSalesforceBaseUrl().toExternalForm() + '/TestVFPage?AcoountId=999';
    PageReference newUrl = new PageReference(strUrl);
    newURL.setRedirect(true);
    return newURL;
}
OR
public PageReference testRedirect() {
    String strUrl = URL.getOrgDomainUrl().toExternalForm() + '/TestVFPage?AcoountId=999';
    PageReference newUrl = new PageReference(strUrl);
    newURL.setRedirect(true);
    return newURL;
}
</pre>

